### ISSUE TITLE
DataInheritance cop similar to StructInheritance

### ISSUE THREAD

**Issue by bdewater** [rubocop/rubocop#11696]
## Is your feature request related to a problem? Please describe.

https://github.com/rubocop/rubocop/blob/master/lib/rubocop/cop/style/struct_inheritance.rb exists because using the bad form creates a useless anonymous class ([as documented](https://ruby-doc.org/core-2.7.0/Struct.html#:~:text=Subclassing%20an%20anonymous%20struct%20creates%20an%20extra%20anonymous%20class%20that%20will%20never%20be%20used.)):
```
irb(main):001:0> class MyStruct < Struct.new(:foo); end
=> nil
irb(main):002:0> MyStruct.ancestors
=> [MyStruct, #<Class:0x0000000105e95790>, Struct, Enumerable, Object, PP::ObjectMixin, Kernel, BasicObject]
```

[Ruby 3.2's Data class](https://ruby-doc.org/3.2.1/Data.html) has the same issue, so it makes sense to have that cop (or a new cop) detect the same pattern:

```
irb(main):003:0> class Point < Data.define(:x, :y); end
=> nil
irb(main):004:0> Point.ancestors
=> [Point, #<Class:0x0000000105e9ef70>, Data, Object, PP::ObjectMixin, Kernel, BasicObject]
```

## Describe the solution you'd like

```ruby
# bad
class Point < Data.define(:x, :y); end

# good
Point = Data.define(:x, :y)
```

## Describe alternatives you've considered

N/A

## Additional context

N/A

**Comment by ktopolski** [comment]
Hey 👋 

If this issue is free for grabs I'd love to work on it! Been using rubocop for years and would love to contribute back to the community 😅

**Comment by bdewater** [comment]
Go for it!

**Comment by MatheusRich** [comment]
@koic @bdewater I wonder if we should revert this or improve it.

Using inheritance is a very straightforward way to create constants within a class:

```rb
class MyClass < Data.define(:foo, :bar)
  A_CONST = 1
end
```
